File: src/7.reverseinteger.py
class Solution(object):
  
  def reverse(self, x):
    neg = False
    result = 0
    if x < 0:
      neg = True
      x = -x
    
    while x > 0:
      digit = x % 10
      x = x - digit
      x = x / 10
      result = result * 10 + digit
      
    if neg:
      result = -result
    if result > 2147483647 or result < -2147483648:
        return 0
    return result

File: src/7.reverseinteger.java
public class Solution {
    public int reverse(int x) {
        boolean pos = true;
        if(x == Integer.MIN_VALUE){
            return 0;
        }
        if(x < 0){
            pos = false;
            x = x * -1;
        }
        long newNum = 0;
        while(x > 0){
            int cur = x%10;
            newNum = newNum * 10 + cur;
            if(newNum > Integer.MAX_VALUE) return 0;
            x = x/10;
        }
        if(pos){
            return (int) newNum;
        }
        else{
            return -(int) newNum;
        }
    }
}